components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java

```diff
@@ -605,6 +607,11 @@ private List<AuthenticatorListItem> buildAuthenticatorListResponse(
                 authenticatorListItem.setName(config.getName());
                 authenticatorListItem.setDisplayName(config.getDisplayName());
                 authenticatorListItem.setIsEnabled(config.isEnabled());
+                if (config.getProperties().length > 0 && IS_API_BASED_SUPPORTED.equals
```

Comment:
Lets do null check here if there is any possibility of config.getProperties() getting null